samcli/lib/init/default_samconfig.py

```diff
@@ -0,0 +1,80 @@
+"""
+Module for creating default samconfig.toml files after initialize a sample app
+"""
+from enum import Enum
+from dataclasses import dataclass
+from typing import Any, List
+
+from samcli.lib.config.samconfig import SamConfig, DEFAULT_GLOBAL_CMDNAME
+from samcli.lib.utils.packagetype import IMAGE, ZIP
+
+MORE_INFO_COMMENT = """More information about the configuration file can be found here:
+# https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-config.html"""
+
+
+class WritingType(Enum):
+    ZIP = ZIP
+    Image = IMAGE
+    Both = "BOTH"
+
+
+@dataclass
+class Default:
+    writing_type: WritingType
+    key: str
+    value: Any
+    command: List[str]
+
+
+class DefaultSamconfig:
+    def __init__(self, path, package_type, project_name):
+        self._path = path
+        self._package_type = package_type
+        self._project_name = project_name
+        self._config = SamConfig(path)
+        self._defaults = [
+            Default(
+                writing_type=WritingType.Both,
+                key="stack_name",
+                value=f"{self._project_name}",
+                command=[f"{DEFAULT_GLOBAL_CMDNAME}"],
+            ),
+            Default(writing_type=WritingType.ZIP, key="cached", value=True, command=["build"]),
+            Default(writing_type=WritingType.Both, key="parallel", value=True, command=["build"]),
```

Comment:
not sue if we should make parallel, and cached as default values for sam build?

Reply:
Generally should be good. Users typically build off the init templates and add new functionality as see they fit. So this will ensure performance is good as they iterate.